snuba/web/views.py

```diff
@@ -92,23 +94,33 @@ def check_clickhouse() -> bool:
     Checks if all the tables in all the enabled datasets exist in ClickHouse
     """
     try:
-        for name in get_enabled_dataset_names():
-            dataset = get_dataset(name)
-            for entity in dataset.get_all_entities():
-                for storage in entity.get_all_storages():
-                    clickhouse = storage.get_cluster().get_query_connection(
-                        ClickhouseClientSettings.QUERY
-                    )
-                    clickhouse_tables = clickhouse.execute("show tables")
-                    source = storage.get_schema()
-                    if isinstance(source, TableSchema):
-                        table_name = source.get_table_name()
-                        if (table_name,) not in clickhouse_tables:
-                            return False
+        datasets = [get_dataset(name) for name in get_enabled_dataset_names()]
+        entities = itertools.chain(
+            *[dataset.get_all_entities() for dataset in datasets]
+        )
+        storages = list(
+            itertools.chain(*[entity.get_all_storages() for entity in entities])
+        )
+        known_table_names = [
+            cast(TableSchema, storage.get_schema()).get_table_name()
+            for storage in storages
+            if isinstance(storage.get_schema(), TableSchema)
+        ]
+        unique_clusters = set([storage.get_cluster() for storage in storages])
```

Comment:
I doubt it causes an issue but these seem like they could be memoized as I wouldn't expect any of these to change without a deploy

Reply:
Definitely could.

```
import functools

@functools.lru_cache
def get_unique_clickhouse_clusters():
    # your function
    
```

But yes the performance issue is not an issue

Reply:
This is not enough to identify uniquer clusters. 
You may have separate cluster config that use the same proxy host on Envoy.
For example these clusters: `erorrs_ro`, `errors`, `cdc` are all accessed through `localhost:WHATEVER`. 

So what you really want to deduplicate are host:port pairs and not the entire cluster config. I can point you to the prod config offline